core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages-store/dot-pages.store.ts

```diff
@@ -0,0 +1,128 @@
+            return {
+                ...state,
+                favoritePages: {
+                    ...state.favoritePages,
+                    items: [...favoritePages]
+                }
+            };
+        }
+    );
+
+    // EFFECTS
+    readonly loadAllFavoritePages = this.effect<void>((trigger$: Observable<unknown>) =>
+        trigger$.pipe(
+            switchMap(() =>
+                this.dotESContentService
+                    .get({
+                        itemsPerPage: this.favoritePagesTotalSize - this.initialFavoritePagesLimit,
+                        offset: this.initialFavoritePagesLimit.toString(),
+                        query: FAVORITE_PAGES_ES_QUERY,
+                        sortField: 'dotFavoritePage.order',
+                        sortOrder: ESOrderDirection.ASC
+                    })
+                    .pipe(
+                        tapResponse(
+                            (items) => {
+                                this.favoritePages = [
+                                    ...this.favoritePages.slice(0, this.initialFavoritePagesLimit),
+                                    ...items.jsonObjectView.contentlets
+                                ];
+                                this.setFavoritePages(this.favoritePages);
+                            },
+                            (error: HttpErrorResponse) => {
+                                return this.httpErrorManagerService.handle(error);
+                            }
+                        )
+                    )
+            )
+        )
+    );
+
+    constructor(
+        private dotCurrentUser: DotCurrentUserService,
+        private httpErrorManagerService: DotHttpErrorManagerService,
+        private dotESContentService: DotESContentService
+    ) {
+        super(null);
+    }
+
+    /**
+     * Sets initial state data from props, roles and current logged user data
+     *
+     * @memberof DotFavoritePageStore
+     */
+    setInitialStateData(): void {
+        forkJoin([
+            this.dotESContentService.get({
+                itemsPerPage: this.initialFavoritePagesLimit,
+                offset: '0',
+                query: FAVORITE_PAGES_ES_QUERY,
+                sortField: 'dotFavoritePage.order',
+                sortOrder: ESOrderDirection.ASC
+            }),
+            this.dotCurrentUser.getCurrentUser()
+        ])
+            .pipe(take(1))
+            .subscribe(([favoritePages, currentUser]: [ESContent, DotCurrentUser]): void => {
+                this.favoritePages = favoritePages?.jsonObjectView.contentlets;
+                this.favoritePagesTotalSize = favoritePages?.resultsSize;
+
+                this.setState({
+                    favoritePages: {
+                        items: favoritePages?.jsonObjectView.contentlets,
+                        total: this.favoritePagesTotalSize
+                    },
+                    loggedUserId: currentUser.userId
+                });
```

Comment:
You're keeping a double state here, one internal and one external (which is consumed by the component), is an unnecessary extra layer of complexity.

Reply:
refactored